### PR TITLE
Rename whitelist/blacklist to allowlist/blocklist

### PR DESCRIPTION
**Pull request by nicodv**
**Related issues**
N/A

**Describe the proposed solution**
N/A

**Describe alternatives you've considered**
N/A

**Additional context**
This lets Transmogrifai adhere to Salesforce's Inclusive Content initiative. Backwards compatible for models that still use old terms.

### CODE REVIEW

File: core/src/main/scala/com/salesforce/op/OpWorkflowModelWriter.scala [hunk 1]
```diff
@@ -165,9 +165,9 @@ private[op] object OpWorkflowModelReadWriteShared {
     val values = findValues
     case object Uid extends FieldNames("uid")
     case object ResultFeaturesUids extends FieldNames("resultFeaturesUids")
-    case object BlacklistedFeaturesUids extends FieldNames("blacklistedFeaturesUids")
-    case object BlacklistedMapKeys extends FieldNames("blacklistedMapKeys")
-    case object BlacklistedStages extends FieldNames("blacklistedStages")
+    case object BlocklistedFeaturesUids extends FieldNames("blocklistedFeaturesUids")
```
Comment (tovbinm) [inline core/src/main/scala/com/salesforce/op/OpWorkflowModelWriter.scala:168]: This change would any existing models with a blacklist defined behave incorrectly. I would recommend to keep backwards compatibility of `resolveBlocklist` (fka `resolveBlacklist`) at least until the next major release, i.e. 0.8.0.

Comment (nicodv) [reply core/src/main/scala/com/salesforce/op/OpWorkflowModelWriter.scala:168]: Good point. Let me try to keep backwards compatibility then. Marking as WIP for now.

Comment (nicodv) [reply core/src/main/scala/com/salesforce/op/OpWorkflowModelWriter.scala:168]: I've added backwards compatibility and a test for it, @tovbinm

File: core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala [hunk 2]
```diff
@@ -166,31 +167,44 @@ class OpWorkflowModelReader(val workflowOpt: Option[OpWorkflow]) extends MLReade
...
+    }.maxBy(_.getOrElse(Array()).length)
   }
 
-  private def resolveBlacklistMapKeys(json: JValue): Try[Map[String, Set[String]]] = Try {
-    (json \ BlacklistedMapKeys.entryName).extractOpt[Map[String, List[String]]] match {
-      case Some(blackMapKeys) => blackMapKeys.map { case (k, vs) => k -> vs.toSet }
-      case None => Map.empty
-    }
+  private def resolveBlocklistMapKeys(json: JValue): Try[Map[String, Set[String]]] = Try {
+    // For backward compatibility. The relevant field name is determined
```
Comment (tovbinm) [inline core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala:199]: 5c: you have `extractOpt`, does it not return `Option[Map[...]]`?

Comment (nicodv) [reply core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala]: Indeed. Using `flatMap` now.

File: helloworld/src/main/resources/BostonDataset/housing.names [hunk 3]
```diff
@@ -41,12 +41,9 @@
     9. RAD       index of accessibility to radial highways
     10. TAX      full-value property-tax rate per $10,000
     11. PTRATIO  pupil-teacher ratio by town
-    12. B        1000(Bk - 0.63)^2 where Bk is the proportion of blacks 
-                 by town
+    12. AA       1000(x - 0.63)^2 where x is the proportion of
```
[the remark this answers is not in the document]
Comment (nicodv) [reply helloworld/src/main/resources/BostonDataset/housing.names:44]: The fact that this variable is available is perhaps a bigger issue than the complicated question of terminology. Let me take this out of the PR and we can tackle the Boston data set separately.

For reference, Scikit-learn decided to remove this data set altogether: https://github.com/scikit-learn/scikit-learn/issues/16155

**Pull request merged**